lens/sqlrepo/repo.go

```diff
@@ -51,5 +54,54 @@ func NewAPIOpener(c *cli.Context) (lens.APIOpener, lens.APICloser, error) {
 		return &tsk, nil
 	}
 
-	return util.NewAPIOpener(c.Context, pgbs, getHeadWithOffset, c.Int("lens-cache-hint"))
+	return util.NewAPIOpener(c.Context, &wrapper{pg: pgbs}, getHeadWithOffset, c.Int("lens-cache-hint"))
+}
+
+type wrapper struct {
+	pg *pgchainbs.PgBlockstore
+}
+
+func (w *wrapper) DeleteBlock(c cid.Cid) error {
+	return w.pg.DeleteBlock(c)
+}
+
+func (w *wrapper) Has(c cid.Cid) (bool, error) {
+	return w.pg.Has(c)
+}
+
+func (w *wrapper) Get(c cid.Cid) (blocks.Block, error) {
+	return w.pg.Get(c)
+}
+
+func (w *wrapper) GetSize(c cid.Cid) (int, error) {
+	return w.pg.GetSize(c)
+}
+
+func (w *wrapper) Put(blk blocks.Block) error {
+	return w.pg.Put(blk)
+}
+
+func (w *wrapper) PutMany(blks []blocks.Block) error {
+	return w.pg.PutMany(blks)
+}
+
+func (w *wrapper) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
+	return w.pg.AllKeysChan(ctx)
+}
+
+func (w *wrapper) HashOnRead(enabled bool) {
+	w.pg.HashOnRead(enabled)
+}
+
+func (w *wrapper) View(c cid.Cid, callback func([]byte) error) error {
+	return w.pg.View(c, callback)
+}
+
+func (w *wrapper) DeleteMany(cids []cid.Cid) error {
+	for _, c := range cids {
+		if err := w.pg.DeleteBlock(c); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
Wrapper needed since DeleteMany isn't implemented left comment inquiring about this: https://github.com/filecoin-project/go-bs-postgres-chainnotated/pull/1/files#r600912738